Add configurable OpenAI base URL support

SmartManoj opened the pull request:
https://github.com/SmartManoj/tetra/pull/1

Copilot committed:
Initial plan

Copilot committed:
Initial analysis: Understanding current OpenAI implementation

Co-authored-by: SmartManoj <7231077+SmartManoj@users.noreply.github.com>

Copilot committed:
Add OpenAI base URL configuration support

Co-authored-by: SmartManoj <7231077+SmartManoj@users.noreply.github.com>

SmartManoj committed:
Merge branch 'master' into copilot/fix-4954cab3-bca2-4ec9-8f7b-c83c752ff3fb

Copilot committed:
Add configurable AI model name support

Co-authored-by: SmartManoj <7231077+SmartManoj@users.noreply.github.com>

Copilot committed:
Improve HTTP error handling and fix log export permissions

Co-authored-by: SmartManoj <7231077+SmartManoj@users.noreply.github.com>

Copilot committed:
Fix log export to Downloads folder and improve LLM error handling

Co-authored-by: SmartManoj <7231077+SmartManoj@users.noreply.github.com>

Copilot committed:
Fix unresolved reference 'exportLogsToFile' by moving function declaration before usage

Co-authored-by: SmartManoj <7231077+SmartManoj@users.noreply.github.com>

Bentlybro commented on Tetra/app/src/main/AndroidManifest.xml:
```diff
@@ -8,6 +8,11 @@
     <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
+    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
+    <!-- For Android 13+ (API 33+) scoped storage -->
+    <uses-permission android:name="android.permission.READ_MEDIA_IMAGES"/>
+    <uses-permission android:name="android.permission.READ_MEDIA_AUDIO"/>
+    <uses-permission android:name="android.permission.READ_MEDIA_VIDEO"/>
```
Curious what these are needed for? maybe out of scope based on the PR title?

SmartManoj replied:
https://github.com/SmartManoj/tetra/pull/1#issuecomment-3252184086

currently the logs are not exported. But these permissions are not needed.

Bentlybro reviewed (commented):
Other than my one question this PR looks good to me :) pleasse can i get some clarification my comment and then this will be good to test and go from there! 

Also can we update the PR with the latest master? some changes to ci where made

Bentlybro committed:
Merge branch 'master' into copilot/fix-4954cab3-bca2-4ec9-8f7b-c83c752ff3fb

SmartManoj commented:
Will reopen after https://github.com/SmartManoj/tetra/pull/1#issuecomment-3252839027

Copilot committed:
Add test LLM button in settings with loading state and toast messages

Co-authored-by: SmartManoj <7231077+SmartManoj@users.noreply.github.com>

SmartManoj committed:
Change pull_request_target to pull_request event

SmartManoj committed:
Disable Discord notifications in APK build workflow

Removed Discord notifications for build status.

SmartManoj committed:
Disable Discord Build Started Notification

Disable Discord build notification for PRs.

SmartManoj committed:
Fix APK signing configuration for consistent installations

Co-Authored-By: SmartManoj <7231077+SmartManoj@users.noreply.github.com>

Copilot committed:
Fix Test LLM Connection to work with default settings by using simple API call without tools

Co-authored-by: SmartManoj <7231077+SmartManoj@users.noreply.github.com>

Copilot committed:
Fix "No response from LLM" error with enhanced network error handling and debugging

Co-authored-by: SmartManoj <7231077+SmartManoj@users.noreply.github.com>

Copilot committed:
Add Test LLM with Tools button using same function as agent for accurate testing

Co-authored-by: SmartManoj <7231077+SmartManoj@users.noreply.github.com>

Copilot committed:
Improve LLM error handling by removing Sentry wrapper and adding detailed network error messages

Co-authored-by: SmartManoj <7231077+SmartManoj@users.noreply.github.com>

Copilot committed:
Fix LLM test button error messages and add commit hash to About page

Co-authored-by: SmartManoj <7231077+SmartManoj@users.noreply.github.com>

Copilot committed:
Fix network on main thread error by using Dispatchers.IO for LLM test functions

Co-authored-by: SmartManoj <7231077+SmartManoj@users.noreply.github.com>

SmartManoj committed:
Refactor UI notifications and improve LLM test threading

Replaced direct Toast usage in DebugScreen with NotificationHelper for consistent notification handling. Updated MainViewModel to ensure all LLM test result callbacks and state updates are dispatched on the main thread using withContext(Dispatchers.Main). Also improved LLMClient OkHttp client configuration with custom timeouts and clarified baseUrl usage.

SmartManoj committed:
Remove unused media and storage permissions

Deleted READ_EXTERNAL_STORAGE and scoped storage permissions (READ_MEDIA_IMAGES, READ_MEDIA_AUDIO, READ_MEDIA_VIDEO) from AndroidManifest.xml as they are no longer required.
